Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneDefinition/Components/SceneDefinitionComponent.cs

```diff
@@ -9,61 +8,130 @@
-        public readonly IpfsPath IpfsPath;
-        public readonly bool IsEmpty;
-        public readonly bool IsSDK7;
-        public readonly ParcelMathHelper.SceneGeometry SceneGeometry;
-        public int InternalJobIndex;
+        public int InternalJobIndex { get; set; }
 
-        public SceneDefinitionComponent(SceneEntityDefinition definition, IpfsPath ipfsPath)
+        public SceneDefinitionComponent(
+            SceneEntityDefinition definition,
+            IReadOnlyList<Vector2Int> parcels,
+            IReadOnlyList<ParcelMathHelper.ParcelCorners> parcelsCorners,
+            ParcelMathHelper.SceneGeometry sceneGeometry,
+            IpfsPath ipfsPath, bool isEmpty, bool isSDK7, bool isPortableExperience)
         {
             Definition = definition;
-            ParcelsCorners = new List<ParcelMathHelper.ParcelCorners>(definition.metadata.scene.DecodedParcels.Select(ParcelMathHelper.CalculateCorners));
+            Parcels = parcels;
+            ParcelsCorners = parcelsCorners;
             IpfsPath = ipfsPath;
-            Parcels = definition.metadata.scene.DecodedParcels;
-            IsEmpty = false;
-            IsSDK7 = definition.metadata?.runtimeVersion == "7";
-            SceneGeometry = ParcelMathHelper.CreateSceneGeometry(ParcelsCorners, Definition.metadata.scene.DecodedBase);
+            IsEmpty = isEmpty;
+            IsSDK7 = isSDK7;
+            SceneGeometry = sceneGeometry;
             InternalJobIndex = -1;
+            IsPortableExperience = isPortableExperience;
+        }
+    }
+
+    public static class SceneDefinitionComponentFactory
+    {
+        private static ParcelMathHelper.SceneGeometry cachedPortableExperiencesSceneGeometry;
+        private static List<ParcelMathHelper.ParcelCorners> cachedParcelsCorners;
+
+        public static SceneDefinitionComponent CreateFromDefinition(SceneEntityDefinition definition, IpfsPath ipfsPath)
+        {
+            if (definition.metadata.isPortableExperience) { return CreatePortableExperienceSceneDefinitionComponent(definition, ipfsPath); }
+
+            return CreateSceneDefinitionComponent(definition, definition.metadata.scene.DecodedParcels, ipfsPath, isEmpty: false, isSDK7: definition.metadata.runtimeVersion == "7", isPortableExperience: false);
+        }
+
+        public static SceneDefinitionComponent CreatePortableExperienceSceneDefinitionComponent(SceneEntityDefinition definition, IpfsPath ipfsPath)
+        {
+            if (cachedParcelsCorners == null)
+            {
+                var portableParcels = new List<Vector2Int>();
+
+                for (int i = -150; i < 150; i++)
+                {
+                    for (int j = -150; j < 150; j++) { portableParcels.Add(new Vector2Int(i, j)); }
+                }
+
+                cachedParcelsCorners = new List<ParcelMathHelper.ParcelCorners>(portableParcels.Select(ParcelMathHelper.CalculateCorners));
```

Comment:
You don't need Parcel Corners for PX: it is only used by `PartitionSceneEntitiesSystem` and you should exclude PX from the system: its partition should be always `{bucket: 0, isBehind: false}` and you can set it upon creation. Currently you just waste CPU time for so many corners during re-partitioning

Reply:
Turns out I was trying to do this and for some reason the scene would never spawn. So I ended up leaving it there. But the issue was that I was not setting the component as dirty 🤦 and it needs to be dirty to be grabbed by the ResolveVisualSceneStateSystem... So thank you for insisting on this, now it works and its excluded from calcualtions!